nixos/modules/services/networking/netbird.nix

```diff
@@ -645,6 +668,77 @@ in
         });
       '';
     })
+    # Setup Keys login automation
+    {
+      systemd.services = pipe cfg.clients [
+        (filterAttrs (_: client: client.login.enable))
+        (mapAttrs' (
+          _: client:
+          nameValuePair "${client.service.name}-login" {
+            after = [ "${client.service.name}.service" ] ++ client.login.systemdDependencies;
+            requires = [ "${client.service.name}.service" ] ++ client.login.systemdDependencies;
+            wantedBy = [ "${client.service.name}.service" ];
+
+            serviceConfig = {
+              Type = "oneshot";
+              RemainAfterExit = true;
+
+              User = client.user.name;
+              Group = client.user.group;
+
+              RemoveIPC = true;
+              PrivateTmp = true;
+              ProtectSystem = "strict";
+              ProtectHome = "yes";
+
+              LoadCredential = [ "setup-key:${client.login.setupKeyFile}" ];
+            };
+
+            environment.NB_SETUP_KEY_FILE = "%d/setup-key";
+            /*
+              might want to do something similar to the docker entrypoint (watching log messages) instead
+              see might want to do something similar to https://github.com/netbirdio/netbird/blob/dc30dcacce4c322502975f1f491e6774efd7e1e9/client/netbird-entrypoint.sh
+            */
+            script = ''
+              set -x
+              nb='${lib.getExe client.wrapper}'
+              keyFile="''${NB_SETUP_KEY_FILE:-"/run/credentials/${client.service.name}.service/setup-key"}"
+
+              fetch_status() {
+                status="$("$nb" status 2>&1)"
+              }
+
+              print_status() {
+                test -n "$status" || refresh_status
+                cat <<EOF
+              $status
+              EOF
+              }
+
+              refresh_status() {
+                fetch_status
+                print_status
+              }
+
+              main() {
+                print_status | ${lib.getExe pkgs.gnused} 's/^/STATUS:INIT: /g'
+                while refresh_status | grep --quiet 'Disconnected' ; do
+                  sleep 1
+                done
+                print_status | ${lib.getExe pkgs.gnused} 's/^/STATUS:WAIT: /g'
+
+                if print_status | grep --quiet 'NeedsLogin' ; then
+                  echo "Using Setup Key File with key: $(cut -b1-8 <"$keyFile")" >&2
+                  "$nb" up --setup-key-file="$keyFile"
+                fi
+              }
+
+              main "$@"
```

Comment:
We don't do anything with the args, also wrapping this in a function for me just makes things more complicated. 😅